tests/integration/test_http_get_all.py

```diff
@@ -0,0 +1,93 @@
+    * Check that filtering parameters are working properly (page, per_page, and state)
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    requests_amount = test_env["http_get_all"]["requests_amount"]
+    submitted_requests = []
+    for _ in range(requests_amount):
+        initial_response = client.create_new_request(
+            payload={
+                "repo": test_env["package"]["repo"],
+                "ref": test_env["package"]["ref"],
+                "pkg_managers": test_env["package"]["pkg_managers"],
+            },
+        )
+        completed_response = client.wait_for_complete_request(initial_response)
+        response_data = completed_response.data
+        submitted_requests.append(response_data.get("id"))
+
+    assert_completed_requests(client, requests_amount)
+    assert_no_requests_in_progress_state(client, requests_amount, submitted_requests)
+
+
+def test_get_all_verbose(test_env, default_request, tmpdir):
+    """
+    Check HTTP GET_ALL with verbose works as expected.
+
+    Process:
+    * Create a request and wait for it to complete
+    * Send request get_all to the Cachito API with verbose
+    Checks:
+    * Check that response code is 200
+    * Check that verbose is working properly
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    assert_verbose_correct(client, test_env)
+
+
+def assert_completed_requests(client: utils.Client, requests_amount):
+    """Check get_all request returns expected completed values."""
+    request_json = {"per_page": requests_amount, "page": 1, "state": "complete"}
+    response = client.fetch_all_requests(request_json)
+    assert response.status == 200
+
+    assert all(item["state"] == "complete" for item in response.data.get("items"))
+    assert 0 <= len(response.data.get("items")) <= requests_amount
+
+
+def assert_no_requests_in_progress_state(client: utils.Client, requests_amount, submitted_requests):
+    """Check get_all request returns expected in_progress values."""
+    request_json = {"per_page": requests_amount, "page": 1, "state": "in_progress"}
+    response = client.fetch_all_requests(request_json)
+    assert response.status == 200
+
+    assert all(item["state"] == "in_progress" for item in response.data.get("items"))
+
+    response_ids = set(item.get("id") for item in response.data.get("items"))
+    assert not any(request_id in response_ids for request_id in submitted_requests)
+    assert 0 <= len(response.data.get("items")) <= requests_amount
+
+
+def assert_verbose_correct(client, test_env):
+    """Check get_all request with verbose=True has expected data."""
+    initial_response = client.create_new_request(
+        payload={
+            "repo": test_env["package"]["repo"],
+            "ref": test_env["package"]["ref"],
+            "pkg_managers": test_env["package"]["pkg_managers"],
+        },
+    )
+    client.wait_for_complete_request(initial_response)
```

Comment:
Could you move this to `test_get_all_verbose`?

Reply:
Done, thanks!